src/linear_algebra/projective_space/basic.lean

```diff
@@ -0,0 +1,232 @@
+begin
+  dsimp only [projectivization.submodule],
+  obtain ⟨a, ha, h⟩ := exists_smul_eq_mk_rep K v hv,
+  rw ← h,
+  apply submodule.span_singleton_smul_eq _ ha,
+end
+
+lemma finrank_submodule (v : projectivization K V) : finrank K v.submodule = 1 :=
+finrank_span_singleton v.rep_nonzero
+
+instance (v : projectivization K V) : finite_dimensional K v.submodule :=
+by { dsimp [projectivization.submodule], apply_instance }
+
+lemma submodule_injective : function.injective
+  (projectivization.submodule : projectivization K V → submodule K V) :=
+begin
+  intros u v h, replace h := le_of_eq h,
+  erw submodule.le_span_singleton_iff at h,
+  rw [← mk_rep v, ← mk_rep u],
+  symmetry,
+  exact quotient.sound' (h u.rep (submodule.mem_span_singleton_self _))
+end
+
+variables (K V)
+/-- The equivalence between the projectivization and the
+collection of subspaces of dimension 1. -/
+noncomputable
+def equiv_submodule : projectivization K V ≃ { H : submodule K V // finrank K H = 1 } :=
+equiv.of_bijective (λ v, ⟨v.submodule, v.finrank_submodule⟩)
+begin
+  split,
+  { intros u v h, apply_fun (λ e, e.val) at h,
+    apply submodule_injective h },
+  { rintros ⟨H, h⟩,
+    rw finrank_eq_one_iff' at h,
+    obtain ⟨v, hv, h⟩ := h,
+    have : (v : V) ≠ 0 := λ c, hv (subtype.coe_injective c),
+    use mk K v this,
+    symmetry,
+    ext x, revert x, erw ← set.ext_iff, ext x,
+    dsimp,
+    rw [submodule_mk, submodule.span_singleton_eq_range],
+    refine ⟨λ hh, _, _⟩,
+    { obtain ⟨c,hc⟩ := h ⟨x,hh⟩,
+      exact ⟨c, congr_arg coe hc⟩ },
+    { rintros ⟨c,rfl⟩,
+      refine submodule.smul_mem _ _ v.2 } }
+end
+variables {K V}
+
+/-- Construct an element of the projectivization from a subspace of dimension 1. -/
+noncomputable
+def mk'' (H : _root_.submodule K V) (h : finrank K H = 1) : projectivization K V :=
+(equiv_submodule K V).symm ⟨H,h⟩
+
+@[simp]
+lemma submodule_mk'' (H : _root_.submodule K V) (h : finrank K H = 1) :
+  (mk'' H h).submodule = H :=
+begin
+  suffices : (equiv_submodule K V) (mk'' H h) = ⟨H,h⟩, by exact congr_arg coe this,
+  dsimp [mk''],
+  simp
+end
+
+@[simp]
+lemma mk''_submodule (v : projectivization K V) : mk'' v.submodule v.finrank_submodule = v :=
+show (equiv_submodule K V).symm (equiv_submodule K V _) = _, by simp
+
+section map
+
+variables {L W : Type*} [field L] [add_comm_group W] [module L W]
+
+/-- A semilinear map of vector spaces induces a map on projective spaces. -/
+def map {σ : K →+* L} (f : V →ₛₗ[σ] W) (hf : function.injective f) :
+  projectivization K V → projectivization L W :=
+quotient.map' (λ v, ⟨f v, λ c, v.2 (hf (by simp [c]))⟩)
+begin
+  rintros ⟨u,hu⟩ ⟨v,hv⟩ ⟨a,ha⟩,
+  use σ a,
+  dsimp at ⊢ ha,
+  rw [← f.map_smulₛₗ, ha],
+end
+
+/-- Mapping with respect to a semilinear map over an isomorphism of fields yields
+an injective map on projective spaces. -/
+lemma map_injective {σ : K ≃+* L} (f : V →ₛₗ[σ.to_ring_hom] W) (hf : function.injective f) :
+  function.injective (map f hf) :=
+begin
```

Comment:
Is there a lemma to be extracted here about when `quotient.map'` is injective?

Reply:
I think this is a somewhat special case for the current context (note e.g. that if \sigma is not an isomorphism then there is no guarantee that `map` will be injective!).